android/src/main/java/pmstation/plantower/USBService.java

```diff
@@ -182,7 +184,7 @@ public boolean wakeUp() {
     private void requestUserPermission() {
         if (!requested) {
             requested = true;
-            PendingIntent pendingIntent = PendingIntent.getBroadcast(this, 0, new Intent(ACTION_USB_PERMISSION), 0);
+            PendingIntent pendingIntent = PendingIntent.getBroadcast(this, 0, new Intent(ACTION_USB_PERMISSION), PendingIntent.FLAG_IMMUTABLE);
```

Comment:
Here also I'm not sure, but I guess IMMUTABLE is ok. @sanchin  can you advice?

Reply:
IMMUTABLE is the correct one here, though I can't remember what 0 meant. Ordinarily you would pass "IMMUTABLE | flag" so that it's ORed resulting in both being passed. Since the old one is 0 I think you can skip it, as long as it doesn't require separate code for older devices (but compiler would complain if it did)